faker/proxy.py

```diff
@@ -235,3 +243,44 @@ def factories(self):
 
     def items(self):
         return self._factory_map.items()
+
+
+class UniqueProxy:
+    def __init__(self, proxy):
+        self._proxy = proxy
+        self._seen = {}
+        self._sentinel = object()
+
+    def clear(self):
+        self._seen = {}
+
+    def __getattr__(self, name: str):
+        obj = getattr(self._proxy, name)
+        if callable(obj):
+            return self._wrap(name, obj)
+        else:
+            return obj
```

Comment:
I am not sure whether this should just throw an exception instead, given you shouldn't be looking up any attribute under `faker.unique` other than a function. Technically, currently, you could recurse. `faker.unique.unique.unique`...

Reply:
Let's throw an exception